src/index.ts

```diff
@@ -116,6 +123,32 @@ class Server {
 
     app.providerStatus = {}
 
+    // feature registration
+    app.apiList = []
+
+    app.registerFeature = (type: string, id: string) => {
+      if (type === 'api' && !app.apiList.includes(id)) {
```

Comment:
should multiple register calls result in error?

Reply:
Now returns true on success / false on error.